pkg/diagnostics/systemd/locate_units.go

```diff
@@ -3,15 +3,35 @@ package systemd
 import (
 	"fmt"
 	"os/exec"
+	"runtime"
 	"strings"
 
 	"github.com/openshift/origin/pkg/diagnostics/log"
 	"github.com/openshift/origin/pkg/diagnostics/types"
 )
 
+// IsSystemd checks if this is a systemd host. We do this by checking that both systemctl and journald are
+// present and functional. When running in a container it is possible for systemctl to be
+// present but non-functional, or misleadingly functional due to sys mounts and a privileged container.
+func IsSystemd() bool {
+	if runtime.GOOS == "linux" {
+		systemctlErr := exec.Command("systemctl").Run()
+		journalctlErr := exec.Command("journalctl", "-n", "1").Run()
```

Comment:
That makes sense thanks, I will split them up. I also discovered we have an env variable we set when in a container so I might skip both if these checks if that is detected as well, which works around the weird systemctl working against host in some containers.